### PR TITLE
interfaces/interfaces/scsi_generic: add interface for scsi generic de…

### PR DESCRIPTION
**Pull request by kubiko**
add interface for scsi generic devices

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

File: interfaces/builtin/scsi_generic.go
Comment (anonymouse64) [inline interfaces/builtin/scsi_generic.go:40]: I think this rule is fine

File: interfaces/builtin/scsi_generic.go
Comment (pedronis) [inline interfaces/builtin/scsi_generic.go:22]: this looks odd, maybe instead:

`allows access to SCSI generic driver devices`

File: interfaces/builtin/scsi_generic.go
Comment (pedronis) [inline interfaces/builtin/scsi_generic.go:30]: shouldn't it be as constrained on the plug side as block-devices?

File: interfaces/builtin/scsi_generic.go
Comment (kubiko) [reply interfaces/builtin/scsi_generic.go:22]: Updated as proposed

File: interfaces/builtin/scsi_generic.go
Comment (pedronis) [reply interfaces/builtin/scsi_generic.go:30]: but as we know from the use case there's can be almost anything behind these devices, while optical-drive are assumed external. So there can be much larger effects writing  to these. So I think we need to be cautious and assume it's more like block-devices than optical-drive.

Comment (kubiko) [reply interfaces/builtin/scsi_generic.go:30]: Plug restriction has been now added.

Comment (anonymouse64) [reply interfaces/builtin/scsi_generic.go:30]: looks correct to me

**Review by anonymouse64 — APPROVED** [review]
lgtm, but I'm not sure if we won't need to modify some tests for the base declaration rules since this is a new super-privileged interface

**Review by mardy — APPROVED** [review]
LGTM!

**Review by pedronis — APPROVED** [review]
lgtm, still needs security review

File: interfaces/builtin/scsi_generic_test.go
Comment (alexmurray) [inline interfaces/builtin/scsi_generic_test.go:4]: Should the Copyright date be 2021?

**Review by alexmurray — APPROVED** [review]
LGTM!

File: interfaces/builtin/scsi_generic_test.go
Comment (anonymouse64) [reply interfaces/builtin/scsi_generic_test.go:4]:
```suggestion
 * Copyright (C) 2021 Canonical Ltd
```

File: interfaces/builtin/scsi_generic.go
Comment (anonymouse64) [inline interfaces/builtin/scsi_generic.go:4]:
```suggestion
 * Copyright (C) 2021 Canonical Ltd
```

**Pull request merged**
